frontend/src/js/components/PageViewer/PageCache.ts

```diff
@@ -119,6 +130,40 @@ export class PageCache {
     };
   };
 
+  refreshPreview = (pageNumber: number, preview: Promise<CachedPreview>, containerSize: number): CachedPage => {  
+    this.setContainerSize(containerSize);
+    const originalPreviewData = this.previewCache.get(pageNumber);
+    const newPreview = updatePreview(preview, containerSize);
+    const newPreviewCache = {
+      previewAbortController: originalPreviewData.previewAbortController,
```

Comment:
This makes me wonder what happens if you zoom while a page preview request is in flight... bit of an edge case but maybe worth testing

Reply:
Great suggestion 👍